diff --git a/configure.ac b/configure.ac
index b127462..fc47412 100644
--- a/configure.ac
+++ b/configure.ac
@@ -105,6 +105,10 @@ AC_COMPILE_IFELSE([AC_LANG_SOURCE([[
  AC_MSG_ERROR([gmp.h can't be found, or is unusable.])
 ])
 
+# look for EGREP and SED here, see continued problem
+# at http://lists.gforge.inria.fr/pipermail/mpc-discuss/2012-January/001056.html
+AC_PROG_EGREP
+AC_PROG_SED
 dnl Check for GMP CFLAGS in gmp.h
 if test -z "$user_redefine_cc" ; then
    MPC_GMP_CC_CFLAGS
diff --git a/m4/mpc.m4 b/m4/mpc.m4
index a014f01..2f68e5d 100644
--- a/m4/mpc.m4
+++ b/m4/mpc.m4
@@ -68,8 +68,6 @@ MPC_GMP_CC_CFLAGS
 AC_DEFUN([MPC_GMP_CC_CFLAGS], [
    AC_MSG_CHECKING(for CC and CFLAGS in gmp.h)
    AC_PROG_CPP
-   AC_PROG_EGREP
-   AC_PROG_SED
    # Get CC
    echo "#include \"gmp.h\"" >  conftest.c
    echo "MPC_OPTION __GMP_CC"           >> conftest.c
